Sources/SwiftSMTP/Mail.swift

```diff
@@ -122,27 +123,72 @@ public struct Mail {
         dictionary["MESSAGE-ID"] = id
         dictionary["DATE"] = Date().smtpFormatted
         dictionary["FROM"] = from.mime
-        dictionary["TO"] = to.map { $0.mime }.joined(separator: ", ")
+        dictionary["TO"] = foldHeaderValue(key: "TO", value: to.map { $0.mime }.joined(separator: ", "))
 
         if !cc.isEmpty {
-            dictionary["CC"] = cc.map { $0.mime }.joined(separator: ", ")
+            dictionary["CC"] = foldHeaderValue(key: "CC", value: cc.map { $0.mime }.joined(separator: ", "))
         }
 
-        dictionary["SUBJECT"] = subject.mimeEncoded ?? ""
+        dictionary["SUBJECT"] = foldHeaderValue(key: "SUBJECT", value: (subject.mimeEncoded ?? ""))
         dictionary["MIME-VERSION"] = "1.0 (Swift-SMTP)"
 
         for (key, value) in additionalHeaders {
             let keyUppercased = key.uppercased()
             if  keyUppercased != "CONTENT-TYPE" &&
                 keyUppercased != "CONTENT-DISPOSITION" &&
                 keyUppercased != "CONTENT-TRANSFER-ENCODING" {
-                dictionary[keyUppercased] = value
+                dictionary[keyUppercased] = foldHeaderValue(key: key, value: value)
             }
         }
 
         return dictionary
     }
 
+    private func foldHeaderValue(key: String, value: String) -> String {
+        let initialHeader = "\(key): \(value)"
+        if initialHeader.count < 79 {
+            return value
+        }
+        // if we're here, it means that RFC 5322 SUGGESTS that we fold this header
+        var foldedHeader = ""
+        var register = "\(key): "
```

Comment:
Won't this result in the key field showing up in the returned string?  I don't believe that's the original behavior.

Reply:
Except that at the end, we remove the key the colon and the whitespace from the beginning of the string. As I noted above, the point here is that the first line has to include the key and the delimiter in the character count. Doing it this way was easier for me to understand, although I suppose one might be clever with arithmetic and advance the linePosition, but we also want to be sure that we don't wind up in a situation like:

```
HEADER: CRLF
 VALUE
```